clang_tools/install.py

```diff
@@ -57,12 +57,15 @@ def clang_tools_binary_url(
     return download_url.replace(" ", "")
 
 
-def install_tool(tool_name: str, version: str, directory: str) -> bool:
+def install_tool(tool_name: str, version: str, directory: str, overwrite: bool) -> bool:
     """An abstract function that can install either clang-tidy or clang-format."""
     if is_installed(tool_name, version):
         # TODO should probably skip this if `directory` is not in the PATH env var.
         print(f"{tool_name}-{version}", "already installed")
-        return False
+        if overwrite:
+            return True
+        else:
+            return False
```

Comment:
The doc for this function is incomplete. The returned value should describe if the binary was downloaded and installed. It should not be used to describe if the the tool was already installed.

The main reason for this is we don't know the full path to an already installed binary. For example, the result of `is_installed()` could be invoking something that was installed via `apt` (not necessarily installed by our tool here).

Reply:
This would lead to bad symlinks or a failure to find the target of the symlink. The `ubunutu-latest` runner already has version 10, 11, and 12 installed (likely via `apt`). This is [shown in the runner's logs](https://github.com/cpp-linter/clang-tools-pip/runs/7803720097?check_suite_focus=true#step:5:14).